tests/app/v2/notifications/test_post_letter_notifications.py

```diff
@@ -0,0 +1,165 @@
+        'personalisation': {'address_line_1': '', 'postcode': ''}
+    }
+
+    error_json = letter_request(client, data, service_id=sample_service.id, _expected_status=400)
+
+    assert error_json['status_code'] == 400
+    assert error_json['errors'] == [{'error': 'BadRequestError', 'message': 'Template not found'}]
+
+
+def test_post_notification_returns_403_and_well_formed_auth_error(
+    client,
+    sample_letter_template
+):
+    data = {
+        'template_id': str(sample_letter_template.id),
+        'personalisation': {'address_line_1': '', 'postcode': ''}
+    }
+
+    error_json = letter_request(client, data, service_id=sample_letter_template.service_id, _expected_status=401)
+
+    assert error_json['status_code'] == 401
+    assert error_json['errors'] == [{
+        'error': 'AuthError',
+        'message': 'Unauthorized, authentication token must be provided'
+    }]
+
+
+def test_notification_returns_400_for_schema_problems(
+    client,
+    sample_service
+):
+    data = {
+        'personalisation': {'address_line_1': '', 'postcode': ''}
+    }
+
+    error_json = letter_request(client, data, service_id=sample_service.id, _expected_status=400)
+
+    assert error_json['status_code'] == 400
+    assert error_json['errors'] == [{
+        'error': 'ValidationError',
+        'message': 'template_id is a required property'
+    }]
+
+
+def test_returns_a_429_limit_exceeded_if_rate_limit_exceeded(
+    client,
+    sample_letter_template,
+    mocker
+):
+    persist_mock = mocker.patch('app.v2.notifications.post_notifications.persist_notification')
+    mocker.patch(
+        'app.v2.notifications.post_notifications.check_rate_limiting',
+        side_effect=RateLimitError('LIMIT', 'INTERVAL', 'TYPE')
+    )
+
+    data = {
+        'template_id': str(sample_letter_template.id),
+        'personalisation': {'address_line_1': '', 'postcode': ''}
+    }
+
+    error_json = letter_request(client, data, service_id=sample_letter_template.service_id, _expected_status=429)
+
+    assert error_json['status_code'] == 429
+    assert error_json['errors'] == [{
+        'error': 'RateLimitError',
+        'message': 'Exceeded rate limit for key type TYPE of LIMIT requests per INTERVAL seconds'
+    }]
+
+    assert not persist_mock.called
+
+
+@pytest.mark.parametrize('service_args', [
+    {'service_permissions': [EMAIL_TYPE, SMS_TYPE]},
+    {'restricted': True}
+])
+def test_post_letter_notification_returns_400_if_not_allowed_to_send_notification(
+    client,
+    notify_db_session,
+    service_args
+):
+    service = create_service(**service_args)
+    template = create_template(service, template_type=LETTER_TYPE)
+
+    data = {
+        'template_id': str(template.id),
+        'personalisation': {'address_line_1': '', 'postcode': ''}
+    }
+
+    error_json = letter_request(client, data, service_id=service.id, _expected_status=400)
+    assert error_json['status_code'] == 400
+    assert error_json['errors'] == [
```

Comment:
We don't currently return 403 for service permissions. See:

https://github.com/alphagov/notifications-api/blob/master/tests/app/v2/notifications/test_post_notifications.py#L293
https://github.com/alphagov/notifications-api/blob/master/tests/app/v2/notifications/test_post_notifications.py#L318
https://github.com/alphagov/notifications-api/blob/master/tests/app/v2/notifications/test_post_notifications.py#L420